repository-data/webfiles/src/main/resources/site/freemarker/common/macro/sections/tableauLookupSection.ftl

```diff
@@ -129,6 +133,7 @@
                         viz${index}.dispose();
                     }
                     viz${index} = new tableau.Viz(viz${index}Elements.vizDiv(), viz${index}Url, options());
+                    document.getElementById('downLoadData').innerHTML = "<br><a href='"+downloadVizLink+"'>Download the coronavirus data for '"+showPostcode+"'</a>";
```

Comment:
Instead of storing showPostcode you should call ' viz${index}Elements.postcode()' to get the snapshot of the value here.

Reply:
good idea, i have removed the redundant showPostcode variable as a consequence.